app/components-react/windows/SourceProperties.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useMemo, useState } from 'react';
+import { Services } from '../service-provider';
+import { ObsForm } from '../obs/ObsForm';
+import { TObsFormData } from '../../components/obs/inputs/ObsInput';
+import { useSelector } from '../store';
+import { ModalLayout } from '../shared/ModalLayout';
+import Display from '../shared/Display';
+import { assertIsDefined } from '../../util/properties-type-guards';
+
+export default function SourceProperties() {
+  const { WindowsService, SourcesService } = Services;
+
+  // get source
+  const source = useMemo(() => {
+    const { sourceId } = WindowsService.getChildWindowQueryParams();
+    return SourcesService.views.getSource(sourceId);
+  }, []);
+
+  // define reactive variables
+  const [properties, setProperties] = useState(source ? source.getPropertiesFormData() : []);
+  const hideStyleBlockers = useSelector(() => WindowsService.state.child.hideStyleBlockers);
+
+  function saveProperties(formData: TObsFormData) {
+    assertIsDefined(source);
+    // save source settings in OBS source
+    source.setPropertiesFormData(formData);
+    // sync source setting with state
+    const updatedProps = source.getPropertiesFormData();
+    setProperties(updatedProps);
+  }
+
+  return (
+    <ModalLayout
+      scrollable
+      fixedChild={source && !hideStyleBlockers && <Display sourceId={source.sourceId} />}
+    >
+      <ObsForm value={properties} onChange={saveProperties} layout="horizontal" />
+    </ModalLayout>
+  );
+}
```

Comment:
This window is missing some functionality that the original window had:
- The window will not automatically close when the underlying source is removed
- The window will not automatically update when the properties are changed programmatically, i.e. when doing an undo/redo operation